### ISSUE TITLE
Paid Stats: Remove credit card entry requirement for free stats purchase

### ISSUE THREAD

**Issue by jsnmoon** [Automattic/wp-calypso#79862]
![Image](https://user-images.githubusercontent.com/4044428/256058293-30ce6e78-311e-439d-9d98-5744ea23573f.png)

Avoid collecting credit cards for free Jetpack Stats plan purchases.

See discussion here: pbNhbs-6ZT-p2#comment-16636.

**Comment by kangzj** [comment]
hi @Automattic/shilling friends, do we have any plan of fixing this? Are we able to pass a parameter or sth to disable credit card entry? Thanks 🤝

**Comment by sirbrillig** [comment]
Yes. There was a comment left on the p2 post (pbNhbs-6ZT-p2#comment-16655) which explains how to do this.

**Comment by DavidRothstein** [comment]
I'm also a little curious what it means to have a $0 product that renews on a yearly basis... if it's free, how and why does it need to renew?  (I'm not sure that process will work smoothly.)

I'm asking mainly because if it were a free **one-time** purchase instead, I believe checkout would already avoid asking for a credit card.  But as noted above, even if you can't do that, the code change required to remove the credit card option directly is pretty simple.

**Comment by kangzj** [comment]
> I'm also a little curious what it means to have a $0 product that renews on a yearly basis... if it's free, how and why does it need to renew? (I'm not sure that process will work smoothly.)

Thanks for the question 🤝  @jboland88  might be a better person to answer the question. 😃

**Comment by kangzj** [comment]
Thanks @sirbrillig for the pointer! We'll work on it 👍

**Comment by jboland88** [comment]
> I'm also a little curious what it means to have a $0 product that renews on a yearly basis... if it's free, how and why does it need to renew? (I'm not sure that process will work smoothly.)

Good question -  I think we just followed the same configuration we used for the Akismet free plans: https://wordpress.com/checkout/akismet/ak_free_yearly
These are also set as annual plans. IIRC, @michaeldcain tested the renewal flow for these and they did not have an issue.

But, that doesn't mean that we could not improve here. The one time purchase sounds like a better solution and would remove the renewal question from the picture. (And it just makes more sense 👍 )

I think in the immediate term, to solve this issue, we can follow the guidance here: pbNhbs-6ZT-p2#comment-16655 (as mentioned above)

I can add a new issue to have a look at replacing the free product(s) with a one-time purchase version. It will likely be a lower priority since everything is working with this setup AFAIK.

**Comment by DavidRothstein** [comment]
>since everything is working with this setup AFAIK

Maybe not working too smoothly though.  I left a comment at https://github.com/Automattic/jetpack-marketing/issues/114 with more details.

>I think in the immediate term, to solve this issue, we can follow the guidance here: pbNhbs-6ZT-p2#comment-16655 (as mentioned above)

That sounds good.  If you do wind up switching to one-time purchases in the future, though, at that point you could probably remove whatever product-specific code you are adding for this now.

**Comment by jeherve** [comment]
This was also suggested here:
https://piaille.fr/@oursinculte/112251687013512035

**Comment by pehaa** [comment]
Mentioned in this thread - https://wordpress.org/support/topic/avoid-use-google-site-kit-instead/

**Comment by pehaa** [comment]
Btw, do I get it right that choosing **Assign a Payment Method Later** allows to proceed without adding a card?

**Comment by sirbrillig** [comment]
> Btw, do I get it right that choosing Assign a Payment Method Later allows to proceed without adding a card?

Correct.

**Comment by jboland88** [comment]
@kangzj - is this still on the radar for your team?

**Comment by kangzj** [comment]
Hi @jboland88 we don’t have any plans to tackle this yet. We are not keen to go too deeply into the checkout page tbh. Might be better job for the payment experts?

**Comment by foleynotrose** [comment]
Can we implement whatever this is, like we use for Search Free: 

<img width="594" alt="Screenshot 2024-05-16 at 10 20 53" src="https://github.com/Automattic/wp-calypso/assets/26651984/f2f96dc5-7d21-4995-8031-3c3c44eac5d6">

(img description: Jetpack Search Free is in the cart, and there is only one Payment Method shown, which is `Free purchase` with a Radio button already toggled on)

**Comment by sirbrillig** [comment]
As [mentioned above](https://github.com/Automattic/wp-calypso/issues/79862#issuecomment-1697641703), it's quite easy to alter checkout to show just the "Free Purchase" payment method and to not collect a card for these purchases. The `allow_free_purchase_method` and `allow_additional_free_purchase_methods` options can be modified for your specific product. It would just require a developer on your team to make the change. There are even unit tests to make sure it works as intended.
